subnet/mock_registry.go

```diff
@@ -68,117 +73,121 @@ func (msr *MockSubnetRegistry) setConfig(network, config string) error {
 	if !ok {
 		return fmt.Errorf("Network %s not found", network)
 	}
-	n.Value = config
+	n.config = config
 	return nil
 }
 
-func (msr *MockSubnetRegistry) getSubnets(ctx context.Context, network string) (*etcd.Response, error) {
+func (msr *MockSubnetRegistry) getSubnets(ctx context.Context, network string) ([]Lease, uint64, error) {
 	n, ok := msr.networks[network]
 	if !ok {
-		return nil, fmt.Errorf("Network %s not found", network)
+		return nil, 0, fmt.Errorf("Network %s not found", network)
 	}
-	return &etcd.Response{
-		Node:  n,
-		Index: msr.index,
-	}, nil
+	return n.subnets, msr.index, nil
 }
 
-func (msr *MockSubnetRegistry) createSubnet(ctx context.Context, network, sn, data string, ttl time.Duration) (*etcd.Response, error) {
+func (msr *MockSubnetRegistry) createSubnet(ctx context.Context, network string, sn ip.IP4Net, attrs *LeaseAttrs, ttl time.Duration) (time.Time, error) {
 	n, ok := msr.networks[network]
 	if !ok {
-		return nil, fmt.Errorf("Network %s not found", network)
+		return time.Time{}, fmt.Errorf("Network %s not found", network)
 	}
 
-	msr.index += 1
-
-	if msr.ttl > 0 {
-		ttl = msr.ttl
+	// check for existing
+	for _, sub := range n.subnets {
+		if sub.Subnet.Equal(sn) {
+			return time.Time{}, etcd.Error{
+				Code:  etcd.ErrorCodeNodeExist,
+				Index: msr.index,
+			}
+		}
```

Comment:
There's at least 4 places here with the "for sub in n.subnets; if Equal(sn)" thing happens; maybe just have a getter that does the search and returns a *Lease?  Then:

if sub := findSubnet(n, sn); sub != nil {
    return time.Time{}, ...
}